echopype/convert/set_groups_azfp.py

```diff
@@ -500,7 +512,18 @@ def set_vendor(self) -> xr.Dataset:
         unpacked_data = self.parser_obj.unpacked_data
         parameters = self.parser_obj.parameters
         ping_time = self.parser_obj.ping_time
-        tdn = parameters["pulse_len"][self.freq_ind_sorted] / 1e6
+        tdn = []
+        for num in parameters["phase_number"]:
+            tdn.append(parameters["pulse_len_phase{}".format(num)][self.freq_ind_sorted] / 1e6)
```

Comment:
I recommend using f string formatting rather than `.format()`. It's typically easier and to read and the intention is clearer. Here's the same line using f string formatting:
```suggestion
            tdn.append(parameters[f"pulse_len_phase{num}"][self.freq_ind_sorted] / 1e6)
```